capsules/src/alarm.rs

```diff
@@ -1,22 +1,19 @@
 //! Provides userspace applications with a alarm API.
 
 use core::cell::Cell;
-use kernel::hil::time::{self, Alarm, Frequency};
+//use kernel::debug;
+use kernel::hil::time::{self, Alarm, Frequency, Ticks};
 use kernel::{AppId, Callback, Driver, Grant, ReturnCode};
 
 /// Syscall driver number.
 use crate::driver;
 pub const DRIVER_NUM: usize = driver::NUM::Alarm as usize;
 
-/// This is based on an exprimental observation (using 16KHz timers) to make sure
-/// that the new tics value is not below the current counter
-/// so the alarm gets fired
-const MIN_TICS_AT_16KHZ: usize = 5;
-
+// This should transition to using Ticks
 #[derive(Copy, Clone, Debug)]
 enum Expiration {
     Disabled,
-    Abs(u32),
+    Enabled(u32, u32), // reference, dt
```

Comment:
Note: I recently discovered that Rust's enumerations allow to use named fields. I think the following would be more readable and less error-prone than having a comment explaining which value is which.

```rust
enum Expiration {
    // ...
    Enabled {
        reference: u32,
        dt: u32,
    }
}
```

Reply:
Good idea.